plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -679,6 +679,138 @@ SELECT
+`
+const sqlAzureDBQueryStoreRuntimeStatistics = sqlAzureDBPartQueryPeriod + `
+
+/*QDS is not enabled on read-only replicas: https://docs.microsoft.com/en-us/sql/t-sql/functions/databasepropertyex-transact-sql?view=sql-server-ver15*/
+IF DATABASEPROPERTYEX(DB_Name(), 'Updatability') = 'READ_ONLY'  
+	RETURN
+
+/* Query the data from query store. Group by plan_id, execution_type, runtime_stats_interval_id, as specified here:
+https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-query-store-runtime-stats-transact-sql?view=sql-server-ver15
+*/
+SELECT
+	'sqlserver_azuredb_querystore_runtime_stats' AS [measurement],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(nvarchar(35), MAX(rsi.start_time), 126) AS interval_start_time,
+	CONVERT(nvarchar(35), MAX(rsi.end_time), 126) AS interval_end_time,
+	CAST(MAX(q.query_id) AS nvarchar(20)) AS query_id,
+	CONVERT(nvarchar(20), MAX(q.query_hash), 1) as query_hash,
+	CAST(rs.plan_id AS nvarchar(20)) AS plan_id,
+	CONVERT(nvarchar(20), MAX(p.query_plan_hash), 1) as query_plan_hash,
+	MAX(rs.max_logical_io_reads) as max_logical_io_reads,
+	SUM(rs.avg_logical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_logical_io_reads,
+	MAX(rs.max_physical_io_reads) as max_physical_io_reads,
+	SUM(rs.avg_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_physical_io_reads,
+	MAX(rs.max_logical_io_writes) as max_logical_io_writes,
+	SUM(rs.avg_logical_io_writes * rs.count_executions) / SUM(rs.count_executions) as avg_logical_io_writes,
+	rs.execution_type,
+	MAX(rs.execution_type_desc) AS execution_type_desc,
+	SUM(rs.count_executions) as count_executions,
+	MAX(rs.max_cpu_time) as max_cpu_time,
+	SUM(rs.avg_cpu_time * rs.count_executions) / SUM(rs.count_executions) as avg_cpu_time,
+	MAX(rs.max_dop) as max_dop,
+	SUM(rs.avg_dop * rs.count_executions) / SUM(rs.count_executions) as avg_dop,
+	MAX(rs.max_rowcount) as max_rowcount,
+	SUM(rs.avg_rowcount * rs.count_executions) / SUM(rs.count_executions) as avg_rowcount,
+	MAX(rs.max_query_max_used_memory) AS max_query_max_used_memory,
+	SUM(rs.avg_query_max_used_memory * rs.count_executions) / SUM(rs.count_executions) AS avg_query_max_used_memory,
+	MAX(rs.max_duration) as [max_duration],
+	SUM(rs.avg_duration * rs.count_executions) / SUM(rs.count_executions) as avg_duration,
+	MAX(rs.max_num_physical_io_reads) as max_num_physical_io_reads,
+	SUM(rs.avg_num_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_num_physical_io_reads,
+	MAX(rs.max_page_server_io_reads) as max_page_server_io_reads,
+	SUM(rs.avg_page_server_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_page_server_io_reads,
+	MAX(rs.max_log_bytes_used) as [max_log_bytes_used],
+	SUM(rs.avg_log_bytes_used * rs.count_executions) / SUM(rs.count_executions) as avg_log_bytes_used
```

Comment:
This is missing a couple of important measures (memory grants and tempdb usage):
1. max_used_memory
2. tempdb_space_used

We should add these.

Reply:
@dimitri-furman, the query already contains field **max_query_max_used_memory**. It looks like what you mentioned. 2 fields related to tempdb were added: **max_tempdb_space_used** and **avg_tempdb_space_used**

Reply:
Yes, I missed `max_query_max_used_memory` being there already, thanks for adding the other one.